BuildResidentialHPXML/resources/schedules.rb

```diff
@@ -908,9 +906,9 @@ def apply_monthly_offsets(array:, weekday_monthly_shift_dict:, weekend_monthly_s
         raise "Could not find the entry for month #{month}, day #{day_of_week} and state #{@state}"
       end
 
-      new_array << array[day * 1440, 1440].rotate(lead)
+      new_array.concat(array[day * 1440, 1440].rotate(lead))
     end
-    return new_array.flatten!
+    return new_array
```

Comment:
Realized we could avoid the flatten call altogether by just building the 1d array from the start. Reduces runtime by ~0.2s.

Reply:
Nice!